server/src/main/java/io/pravega/schemaregistry/service/SchemaRegistryService.java

```diff
@@ -864,6 +873,29 @@ private SchemaInfo normalizeSchemaBinary(SchemaInfo schemaInfo) {
         return new SchemaInfo(schemaInfo.getType(), schemaInfo.getSerializationFormat(), schemaBinary, schemaInfo.getProperties());
     }
 
+    private void validateJsonSchema(String schemaString) {
+        try {
+            // 1. try draft 3
+            // jackson JsonSchema only supports json draft 3. If the schema definition is not compatible with draft 3, 
+            // try parsing the schema with everit library which supports drafts 4 6 and 7. 
+            OBJECT_MAPPER.readValue(schemaString, JsonSchema.class);
+        } catch (IOException e) {
+            handleDraft4onward(schemaString);
+        }
+    }
+
+    private void handleDraft4onward(String schemaString) {
+        JSONObject rawSchema = new JSONObject(new JSONTokener(schemaString));
+        // draft 4 to 7
```

Comment:
Do you mean 4 to 6 here? The condition appears to suggest that as you invoke `draftV7Support()`. 

Also, if you could simplify this condition, while simultaneously making the reason for the condition more explicit:

```
SchemaLoader.SchemaLoaderBuilder schemaLoaderBuilder = SchemaLoader.builder().useDefaults(true).schemaJson(rawSchema);
if (!rawSchema.has("id")) {
       schemaLoaderBuilder.draftV7Support();
}
schemaLoaderBuilder.build().load().build();
```

Reply:
funnily enough they have 4 and 6 and 7. there is no draft 5. 
6,7 and backward compatible. the only incompatible difference between 4 and 7 is "$id" vs "id". 
So if schema has "id" we treat it a 4 otherwise we treat it as 7 which will cover all cases of 4 till 7.

Reply:
With the new comment added in the code, this looks good.